macros/sql/get_tables_by_prefix.sql

```diff
@@ -1,16 +1,24 @@
-{% macro get_tables_by_prefix(schema, prefix, exclude='') %}
+{% macro get_tables_by_prefix(schema, prefix, exclude='', database=target.database) %}
 
-    {%- call statement('tables', fetch_result=True) %}
+    {%- call statement('get_tables', fetch_result=True) %}
 
       {{ dbt_utils.get_tables_by_prefix_sql(schema, prefix, exclude) }}
 
     {%- endcall -%}
 
-    {%- set table_list = load_result('tables') -%}
+    {%- set table_list = load_result('get_tables') -%}
 
-    {%- if table_list and table_list['data'] -%}
-        {%- set tables = table_list['data'] | map(attribute=0) | list %}
-        {{ return(tables) }}
+    {%- if table_list and table_list['table'] -%}
+        {%- set tbl_relations = [] -%}
+        {%- for row in table_list['table'] -%}
+            {%- set tbl_relation = adapter.get_relation(database, row.table_schema, row.table_name) -%}
```

Comment:
What's the rationale behind calling `get_relation` here?

Reply:
I was under the (likely false) impression that, if a relation already exists in dbt's cache, `get_relation` is a less expensive operation than `api.Relation.create`. If these are functionally equivalent, or if the extra lines of code are not worth the minuscule benefit, I'll happily go for the latter

Reply:
I don't think there would be a penalty for calling `get_relation` here (you're right, it should be cached) but I think in this case it's not necessary